app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee

```diff
@@ -0,0 +1,5 @@
+Spree.fetch_recently_viewed_products = (productId = "")->
+  $.ajax
+    url: Spree.pathFor("recently_viewed_products?product_id=" + productId),
```

Comment:
@priyank-gupta we're sending product id to save it also in recently viewed list - that way it is all done in one single request

BTW I see that right now specs are failing - will fix it